lib/msf/ui/web/console.rb

```diff
@@ -51,6 +51,10 @@ def initialize(framework, console_id, opts={})
     # Skip database initialization if it is already configured
     if framework.db && framework.db.active
       opts['SkipDatabaseInit'] = true
+      if opts['workspace']
+        wspace = framework.db.find_workspace(opts['workspace'])
```

Comment:
There's no guarantee that the workspace name that's been passed in exists right?  Out of curiosity do you know what would happen in that case? might not be something worth worrying about

Reply:
Valid question, in this case the find would return `nil` and then set that on the existing framework instance.  That `should` cause an error to be reported on attempt to set a non-existent workspace and report a valid error.  As noted on opening this, workspace behavior is in flux, so this is a good case to keep in mind as chances shake out more.